dotnet/src/SemanticKernel.Skills/Skills.Web/WebFileDownloadSkill.cs

```diff
@@ -73,7 +73,7 @@ public async Task DownloadToFileAsync(string source, SKContext context)
         this._logger.LogDebug("Response received: {0}", response.StatusCode);
 
         using Stream webStream = await response.Content.ReadAsStreamAsync();
-        using FileStream outputFileStream = new FileStream(filePath, FileMode.Create);
+        using FileStream outputFileStream = new FileStream(Environment.ExpandEnvironmentVariables(filePath), FileMode.Create);
```

Comment:
I don't love the idea of expanding variables in a user-provided arg string... shouldn't the caller do this expansion? What's the scenario?

Reply:
I was asked by @craigomatic specifically to add this.  I'll let him speak to this.

Reply:
Our goal with the samples is to enable a new dev to run the samples the first time, without having to edit any code files prior (the identity being an exception here as it needs an AAD client id). Including environment variable expansion in the skill helps to simplify this - the caller in this case is using %temp%\somefile so that it can subsequently unzip the downloaded file.

From a security perspective, this expansion of environment variables does not override any OS level of protection on special folders and if the dev chooses to edit the sample and use a path that should not be written to, they will end up with a SecurityException: 

<img width="539" alt="image" src="https://user-images.githubusercontent.com/146438/225469357-53192cc6-2d5b-4234-a936-1390810bcb73.png">

See also: https://learn.microsoft.com/en-us/dotnet/api/system.environment.getenvironmentvariables?view=net-7.0#system-environment-getenvironmentvariables

Reply:
Maybe we can bypass this whole problem by using @SergeyMenshykh 's suggestion from the first open discussion above.

It would let users choose the name of the file to save to, but not the directory into which it would be saved.  The latter would be configured at the kernel level (with a reasonable default value like %temp%\sk).  We would still expand variables, but not from a user-provided string.

Reply:
This should be taken care of after implementing something like Sergey mentioned.